### PR TITLE
Fix exception from DOB month being a whitespace

### PR DESCRIPTION
**Pull request by benilovj**
### Context

Entering a space instead of a valid month on the 'date of birth' form raises a `Date::Error` exception.

[Sentry](https://sentry.io/organizations/dfe-teacher-services/issues/3274901284/?project=6275068&referrer=slack)

### Changes proposed in this pull request

Strip whitespace on the DOB month.

### Guidance to review

...

### Checklist

- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Review by dankmitchell — APPROVED** [review]
LGTM

**Comment by benilovj** [comment]
Thanks @dankmitchell I'll let the day shift devs review before I merge so they have visibility 😂

**Comment by dankmitchell** [comment]
Wise decision 😅

**Review by tvararu — APPROVED** [review]

**Comment by tvararu** [comment]
:pyjama:

**Pull request merged**
